backend/src/apiserver/filter/filter.go

```diff
@@ -23,9 +23,8 @@ import (
 	"github.com/Masterminds/squirrel"
 	"github.com/golang/protobuf/jsonpb"
 	"github.com/golang/protobuf/ptypes"
-	"github.com/kubeflow/pipelines/backend/src/common/util"
-
 	api "github.com/kubeflow/pipelines/backend/api/v1beta1/go_client"
```

Comment:
Hi @gkcalat, @Linchin 
I already brought up this topic in another PR (see [here](https://github.com/kubeflow/pipelines/pull/8480#issuecomment-1325214714) for context), but I want to comment on it one more time. 
As it has been chosen that the backend is going to use just one of the `filter api` - because `v1.Filtet.proto` and `v2.Filetr.proto` are the same and because they are not directly used in any gRPC call, but serialized encoded (used behind an interface{}) - they are interchangeable - isn't it better to move toward using v2 `github.com/kubeflow/pipelines/backend/api/v2beta1/go_client.Filter` instead of keeping the code depended on `v1.Filter`. Better do it now, instead of later?

Reply:
Hi @difince!
Thank you for taking a look at this PR. This is a valid concern.

If you take a closer look at the changes you can find that many packages are now API-independent. Instead, we convert the incoming messages at the `server` level and pass `model` types down the line. That being said, we haven't finished the refactoring. So ideally, we will end up removing the dependency on the `filter.proto`. If that is not be possible, we might indeed opt out for your proposed solution. Similarly, we will migrate from `v1beta1`'s `auth.proto`, `visualization.proto`, and others to their `v2beta1` counterparts.